[Bug]: Bad `import { composeStories } from '@storybook/react'` in v7.4.0

### Describe the bug

When running tests the `import { composeStories } from "@storybook/react"` fails with:
```
Cannot find module '@storybook/core-events/preview-errors' from 'node_modules/@storybook/preview-api/dist/index.js'
```

With Storybook v7.3.2 test run successfully.

### To Reproduce

Simplified project: **[m-tartari/storybook-issue-24009](https://github.com/m-tartari/storybook-issue-24009)**

use branch `v7.3.2` for passing tests using Storybook v7.3.2

To reproduce the error install dependencies and run the tests:

```shell
npm ci
npm run test -- --watchAll=false
```

or

```shell
yarn install --frozen-lockfile
yarn run test -- --watchAll=false
```

the test fails with:

```
 FAIL  src/ToggleButton/ToggleButton.test.tsx
  ● Test suite failed to run

    Cannot find module '@storybook/core-events/preview-errors' from 'node_modules/@storybook/preview-api/dist/index.js'

    Require stack:
      node_modules/@storybook/preview-api/dist/index.js
      node_modules/@storybook/react/dist/index.js
      src/ToggleButton/ToggleButton.test.tsx

      12 | // Imports a specific story for the test
      13 | import { toggleButtonClasses as classes } from "@mui/material/ToggleButton";
    > 14 | import { composeStories } from "@storybook/react";
         | ^
      15 | import * as stories from "./ToggleButton.stories";
      16 | import userEvent from "@testing-library/user-event";
      17 |

      at Resolver.resolveModule (node_modules/jest-resolve/build/resolver.js:324:11)
      at Object.<anonymous> (node_modules/@storybook/preview-api/dist/index.js:6:402)
      at Object.<anonymous> (node_modules/@storybook/react/dist/index.js:1:1492)
      at Object.<anonymous> (src/ToggleButton/ToggleButton.test.tsx:14:1)
      at TestScheduler.scheduleTests (node_modules/react-scripts/node_modules/@jest/core/build/TestScheduler.js:333:13)
      at runJest (node_modules/react-scripts/node_modules/@jest/core/build/runJest.js:404:19)

Test Suites: 1 failed, 1 total
Tests:       0 total
Snapshots:   0 total
Time:        2.278 s
Ran all test suites.
```

### System

```bash
Environment Info:

  System:
    OS: Linux 5.15 Ubuntu 20.04.6 LTS (Focal Fossa)
    CPU: (8) x64 AMD Ryzen 5 3500U with Radeon Vega Mobile Gfx
  Binaries:
    Node: 18.16.0 - ~/.nvm/versions/node/v18.16.0/bin/node
    Yarn: 1.22.19 - ~/.nvm/versions/node/v18.16.0/bin/yarn
    npm: 9.5.1 - ~/.nvm/versions/node/v18.16.0/bin/npm
  Browsers:
    Chrome: 116.0.5845.140
  npmPackages:
    @storybook/addon-actions: ^7.4.0 => 7.4.0 
    @storybook/addon-docs: ^7.4.0 => 7.4.0 
    @storybook/addon-essentials: ^7.4.0 => 7.4.0 
    @storybook/addon-links: ^7.4.0 => 7.4.0 
    @storybook/addon-storyshots: ^7.4.0 => 7.4.0 
    @storybook/node-logger: ^7.4.0 => 7.4.0 
    @storybook/preset-create-react-app: ^7.4.0 => 7.4.0 
    @storybook/react: ^7.4.0 => 7.4.0 
    @storybook/react-webpack5: ^7.4.0 => 7.4.0 
    @storybook/theming: ^7.4.0 => 7.4.0
```


### Additional context

_No response_

Same error here! Version used: `7.4.0`.

Full error message:
```
Cannot find module '@storybook/core-events/preview-errors' from 'node_modules/@storybook/preview-api/dist/index.js'
```

My package json:
```json
"@storybook/addon-essentials": "^7.4.0",
    "@storybook/addon-interactions": "^7.4.0",
    "@storybook/addon-links": "^7.4.0",
    "@storybook/addon-themes": "^7.4.0",
    "@storybook/addons": "^7.4.0",
    "@storybook/client-api": "^7.4.0",
    "@storybook/jest": "^0.2.2",
    "@storybook/node-logger": "^7.4.0",
    "@storybook/preset-create-react-app": "^7.4.0",
    "@storybook/preview-web": "^7.4.0",
    "@storybook/react": "^7.4.0",
    "@storybook/react-webpack5": "^7.4.0",
    "@storybook/testing-library": "^0.2.0",
    "@storybook/theming": "^7.4.0",
```

`npm why` and `npm explain confirms me that it's all on same versions.

@shilman I don't think I'm getting the wrong version, here's the stack trace:

```
Test suite failed to run

    Cannot find module '@storybook/core-events/preview-errors' from '../../node_modules/.pnpm/@storybook+preview-api@7.4.0/node_modules/@storybook/preview-api/dist/index.js'

    Require stack:
      /Users/justinsmith/Dev/designsystem/node_modules/.pnpm/@storybook+preview-api@7.4.0/node_modules/@storybook/preview-api/dist/index.js
      /Users/justinsmith/Dev/designsystem/node_modules/.pnpm/@storybook+react@7.4.0_react-dom@17.0.2_react@17.0.2_typescript@4.9.3/node_modules/@storybook/react/dist/index.js
      src/setup.ts

      at Resolver.resolveModule (../../node_modules/.pnpm/jest-resolve@27.5.1/node_modules/jest-resolve/build/resolver.js:324:11)
      at Object.<anonymous> (../../node_modules/.pnpm/@storybook+preview-api@7.4.0/node_modules/@storybook/preview-api/dist/index.js:6:402)

```

--- a/code/lib/cli/package.json
+++ b/code/lib/cli/package.json
@@ -61,6 +61,7 @@
     "@ndelangen/get-tarball": "^3.0.7",
     "@storybook/codemod": "workspace:*",
     "@storybook/core-common": "workspace:*",
+    "@storybook/core-events": "workspace:*",
     "@storybook/core-server": "workspace:*",
     "@storybook/csf-tools": "workspace:*",
     "@storybook/node-logger": "workspace:*",

--- a/code/lib/core-common/package.json
+++ b/code/lib/core-common/package.json
@@ -44,6 +44,7 @@
     "prep": "../../../scripts/prepare/bundle.ts"
   },
   "dependencies": {
+    "@storybook/core-events": "workspace:*",
     "@storybook/node-logger": "workspace:*",
     "@storybook/types": "workspace:*",
     "@types/find-cache-dir": "^3.2.1",

--- a/code/lib/core-events/manager-errors.js
+++ b/code/lib/core-events/manager-errors.js
@@ -1,0 +1,4 @@
+// This is required for projects that require paths such as `@storybook/core-events/manager-errors`
+// but in CJS, while not in ESM mode. Else an error like this will occur:
+// ENOENT: no such file or directory, open '/xyz/node_modules/@storybook/core-events/manager-errors.js'
+module.exports = require('./dist/errors/manager-errors');

--- a/code/lib/core-events/preview-errors.js
+++ b/code/lib/core-events/preview-errors.js
@@ -1,0 +1,4 @@
+// This is required for projects that require paths such as `@storybook/core-events/preview-errors`
+// but in CJS, while not in ESM mode. Else an error like this will occur:
+// ENOENT: no such file or directory, open '/xyz/node_modules/@storybook/core-events/preview-errors.js'
+module.exports = require('./dist/errors/preview-errors');

--- a/code/lib/core-events/server-errors.js
+++ b/code/lib/core-events/server-errors.js
@@ -1,0 +1,4 @@
+// This is required for projects that require paths such as `@storybook/core-events/server-errors`
+// but in CJS, while not in ESM mode. Else an error like this will occur:
+// ENOENT: no such file or directory, open '/xyz/node_modules/@storybook/core-events/server-errors.js'
+module.exports = require('./dist/errors/server-errors');

